# Worksheet #6

A1 and Functions

In [1]:
# the usual necessities

%matplotlib inline
import numpy as np
import pylab

Each value in the array below (ages) represents individuals in line for the DMV. 

Groceries is a list of grocery prices at different stores. Each row is an item and each column is a different store. 

In [2]:
dmvline = np.array([79, 92, 12, 25, 37, 62,
                      28, 15, 18, 56, 45, 51,
                      39, 83, 42, 30, 49, 74,
                      33, 49, 21, 44, 48, 88,
                      24, 18, 29, 28, 82, 77,
                      26, 22, 83, 52, 66, 85,
                      55, 37, 71, 73, 83, 35])

groceries = np.array([[1.65, 1.83, 1.94, 2.21, 1.58, 2.04, 2.95],
                     [4.59, 5.69, 4.12, 5.92, 3.99, 4.12, 4.83],
                     [10.5, 10.58, 11.38, 11.85, 10.64, 12.24, 12.00],
                     [0.5, 0.28, 0.49, 0.38, 0.94, 0.72, 1.48],
                     [15.39, 20.38, 18.93, 24.26, 15.36, 14.99, 15.72],
                     [8.28, 7.17, 8.88, 7.27, 7.12, 6.81, 8.00]])

# 1. What is the the maximum increase in age between two consecutive people in line at the DMV?

In [4]:
np.max(np.diff(dmvline))

61

I got 61 for the max

# 2. What is the greatest amount of money someone could spend if they are buying one of each of these items at one store?

In [9]:
np.max(np.sum(groceries, axis=0))

51.890000000000001

I got 51.89

# Functions

# 3. Create a function called area that takes a width and height and returns the area

In [10]:
def area(w, h): 
    return w*h#replace this with some expression
area(6, 5)

30

I got 30

# 4. Create a function called square that takes a number as an argument/parameter and returns the square of that number. Ex: square(5)=25 

In [12]:
def square(x): 
    return x**2#replace this with some expression
square(6)

36

I got 36

# 5. Create a function called fun that takes in an array as a parameter/argument then finds the largest element and smallest element then squares each element (use the function you wrote about) and subtracts the smaller number from the larger number. We will test it with DMV line. 

In [15]:
def fun(x):
    a= np.min(x)
    b= np.max(x)
    return b**2-a**2 #Write your function here

#This statement will test your code do not delete it
fun(dmvline)

8320

I got 8320

# 6. Create a function called transformtime that takes in a number of days until an event and returns the number of hours until the event and the minutes until the event. Ex: transformtime(2) = (48, 2880)
Hint: to return two things simply separate them with a comma after the return statement (ex: return x,y)

In [18]:
def transformtime(x):
    h = x*24
    m = x*24*60
    return (h,m) #Write your function here 

#This statement will test your code do not delete it
transformtime(5)

(120, 7200)

I got (120, 7200)

Submit your notebook below

In [19]:
import comp116
comp116.submit('ws6')